src/main.cpp

```diff
@@ -6038,15 +6040,18 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
     {
         CXThinBlock thinBlock;
         vRecv >> thinBlock;
-        CInv inv(MSG_BLOCK, thinBlock.header.GetHash());
-        // Send expedited ASAP
-        CValidationState state;
-        if (!CheckBlockHeader(thinBlock.header, state, true)) { // block header is bad
-            LogPrint("thin", "Thinblock %s received with bad header from peer %s (%d)\n", inv.hash.ToString(), pfrom->addrName.c_str(), pfrom->id);
-            Misbehaving(pfrom->GetId(), 20);
-            return false;
+
+        // Check that that there is at least one txn in the xthin and that the first txn is the coinbase
```

Comment:
Why comment on what the IsThinBlockValid does? If that method changes you'd have to update this doc in 3 places.